Src/Microsoft.Dynamic/Actions/Calls/TypeInferer.cs

```diff
@@ -414,21 +414,19 @@ class ArgumentInputs {
             if (parameterType.IsSubclassOf(typeof(Delegate))) {
                 // see if we have an invokable object which can be used to infer into this delegate
                 IInferableInvokable invokeInfer = input as IInferableInvokable;
-                if (invokeInfer != null) {
```

Comment:
Hmm, not sure about the pattern matching suggestions, since these are nested checks. Can you provide an example?